engine/consensus/matching/engine_test.go

```diff
@@ -548,25 +548,25 @@ func (ms *MatchingSuite) TestRequestPendingApprovals() {
 	unfinalizedBlock := unittest.BlockWithParentFixture(parentBlock.Header)
 	ms.Blocks[unfinalizedBlock.ID()] = &unfinalizedBlock
 
-	// we will assume that all chunks are assigned to the same verifier. So each
-	// approval request should only be sent to this single verifier.
-	verifier := unittest.IdentifierFixture()
+	// we will assume that all chunks are assigned to the same two verifiers.
+	verifiers := unittest.IdentifierListFixture(2)
+
+	// the matching engine requires approvals from both verifiers for each chunk
+	ms.matching.requiredApprovalsForSealConstruction = 2
 
 	// expectedRequests collects the set of ApprovalRequests that should be sent
 	expectedRequests := []*messages.ApprovalRequest{}
 
 	// populate the incorporated-results mempool with:
-	// - 50 that have collected one signature per chunk
-	// - 50 that have collected no signatures
-	//
-	// each chunk is assigned to the single verifier we defined above
+	// - 50 that have collected two signatures per chunk
+	// - 50 that have collected only 1 signature
```

Comment:
To the best of my understanding, we only test the cases where the incorporated result has 1 or two  approval(s). What about if there are none?  Given that our implementation actually covers this through a special case logic: https://github.com/onflow/flow-go/blob/756629693e3a1c207cad04d9222c76e9a8b9b3a9/engine/consensus/matching/engine.go#L1059 I feel we should also include this in our test.